src/commands/gate/evaluate.ts

```diff
@@ -182,28 +182,39 @@ export class GateEvaluateCommand extends Command {
     const remainingWait = this.timeoutInSeconds * 1000 - timePassed
 
     return new Promise((resolve, reject) => {
-      if (remainingWait <= 0) {
-        bail?.(new Error('wait'))
-      } else {
-        api
-          .evaluateGateRules(evaluateRequest, this.context.stdout.write.bind(this.context.stdout))
-          .then((response) => {
-            if (response.data.data.attributes.status === 'wait') {
-              this.context.stdout.write(renderWaiting())
-              const waitTime = response.data.data.attributes.metadata?.wait_time_ms ?? 0
-              setTimeout(() => {
-                reject(new Error('wait'))
-              }, Math.min(remainingWait, waitTime))
-            } else {
-              resolve(response)
-            }
-          })
-          .catch((err) => {
-            setTimeout(() => {
-              reject(err)
-            }, Math.min(remainingWait, this.getDelay(attempt ?? 1)))
-          })
+      const request = {...evaluateRequest, options: {...evaluateRequest.options}}
+      if (remainingWait <= 0 || attempt === this.maxRetries + 1) {
```

Comment:
Not that we should ever get to `attempt === this.maxRetries + 2`, but would this be more appropriate?
```
attempt > this.maxRetries
```

Reply:
it shouldn't happen ever, if it happens datadog-ci retry is broken.

but if it does, we expect to get a single `lastRetry` in the service (because we'll send the event to EvP, and we want to have 1 event per gate execution), so I'd keep it as is if you don't mind